ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -63,8 +64,21 @@ data IsLedgerPeer = IsLedgerPeer | IsNotLedgerPeer
 
 newtype NumberOfPeers = NumberOfPeers Word16 deriving Show
 
-newtype LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
-      lpGetPeers :: SlotNo -> STM m (Maybe [(PoolStake, NonEmpty RelayAccessPoint)])
+data LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
+      -- | Return ledger peer information from given slot number.
+      --
+      -- The consensus layer functionality that the management of local root
+      -- peers vs ledger peers logic requires.
+      --
+      -- This will be used in order to distribute the load of locally
+      -- configured root peers to ledger peers. However this needs to be done
+      -- carefully, one shouldn't start using ledger peers if it's not safe to
+      -- do so. A good time to switch and start using ledger peers is determined
```

Comment:
> A good time to switch and start using ledger peers is determined by choosing the point so that the probability that a 50% stake adversarial would be chosen k times to produce a block is sufficiently low.

The actual 50% threshold is problematic. There is strictly a >0.5 probability that a 50% adversary will win each race against the honest stake to mint a chain of length k.

So perhaps change the last sentence to

> It is sufficient unlikely that a strong adversary could launch a successful long-range against this victim.